[enhancement] macOS support

So, I tried to build this on macOS and got pretty far, but then ran into some issues, which I thought I would post here.

To setup my macos environment, I used homebrew and ran:

`brew install nuget` 

which installed `mono-6.8.0.96`, and `v5.4.0/nuget.exe`

I then needed to edit one line in the `makefile` so that it read like this (pointing to `/usr/local/lib`):

`     mcs -out:build/steamguard -r:build/SteamAuth.dll -r:build/Newtonsoft.Json.dll -r:/usr/local/lib/mono/4.5/System.Security.dll Program.cs Manifest.cs AssemblyInfo.cs Utils.cs`

At that point `make` works.

```
...
SteamAuth/SteamAuth/SteamGuardAccount.cs(326,20): warning CS0219: The variable `referer' is assigned but its value is never used
SteamAuth/SteamAuth/SteamGuardAccount.cs(346,20): warning CS0219: The variable `referer' is assigned but its value is never used
SteamAuth/SteamAuth/SteamGuardAccount.cs(367,20): warning CS0219: The variable `referer' is assigned but its value is never used
SteamAuth/SteamAuth/UserLogin.cs(71,38): warning CS0219: The variable `secureRandom' is assigned but its value is never used
Compilation succeeded - 4 warning(s)
cp SteamAuth/SteamAuth/packages/Newtonsoft.Json.11.0.1/lib/net45/Newtonsoft.Json.dll build/
mcs -out:build/steamguard -r:build/SteamAuth.dll -r:build/Newtonsoft.Json.dll -r:/usr/local/lib/mono/4.5/System.Security.dll Program.cs Manifest.cs AssemblyInfo.cs Utils.cs
```

However the next build step does not work.

I have never used mono before, so I still need to do some exploration, but I suspect, it is either building for the wrong platform completely or this is a 32 bit application and needs to be build as a 64 bit application on macOS Catalina.

```
$ file build/steamguard
build/steamguard: PE32 executable (console) Intel 80386 Mono/.Net assembly, for MS Windows
```

So, it looks like this wouldn't be that hard to get working. I'd be happy to help out with some testing if you can provide some pointers on what to look for or try.

--- a/README.md
+++ b/README.md
@@ -29,6 +29,34 @@ To run the current build quickly:
 
     make run
 
+## macOS
+
+Building on macOS is pretty simple. Make sure you have all the prerequisites listed above.
+
+* For the prerequisites the easiest way to install them is to use [homebrew](https://brew.sh/):
+
+    brew update
+    brew install mono
+    brew install nuget
+
+* For the application source code:
+
+    git clone --recursive https://github.com/dyc3/steamguard-cli.git
+    cd steamguard-cli
+    make -f makefile.macos
+
+To run the current build:
+
+    mono build/steamguard
+
+To run the current build quickly:
+
+    make -f makefile.macos run
+
+To install run:
+
+    make -f makefile.macos install
+
 ## Windows
 Coming soon...
 

--- a/makefile.macos
+++ b/makefile.macos
@@ -1,0 +1,24 @@
+﻿all: Program.cs
+	mkdir -p build/
+	nuget restore SteamAuth/SteamAuth/SteamAuth.sln
+	mcs -target:library -out:build/SteamAuth.dll -r:SteamAuth/SteamAuth/packages/Newtonsoft.Json.11.0.1/lib/net45/Newtonsoft.Json.dll SteamAuth/SteamAuth/APIEndpoints.cs SteamAuth/SteamAuth/AuthenticatorLinker.cs SteamAuth/SteamAuth/Confirmation.cs SteamAuth/SteamAuth/SessionData.cs SteamAuth/SteamAuth/SteamGuardAccount.cs SteamAuth/SteamAuth/SteamWeb.cs SteamAuth/SteamAuth/TimeAligner.cs SteamAuth/SteamAuth/UserLogin.cs SteamAuth/SteamAuth/Util.cs SteamAuth/SteamAuth/Properties/AssemblyInfo.cs
+	cp SteamAuth/SteamAuth/packages/Newtonsoft.Json.11.0.1/lib/net45/Newtonsoft.Json.dll build/
+	mcs -out:build/steamguard -r:build/SteamAuth.dll -r:build/Newtonsoft.Json.dll -r:/usr/local/lib/mono/4.5/System.Security.dll Program.cs Manifest.cs AssemblyInfo.cs Utils.cs
+
+run:
+	mono build/steamguard -v
+
+clean:
+	rm -r build/
+
+install:
+	cp build/steamguard /usr/local/bin/
+	cp build/Newtonsoft.Json.dll /usr/local/bin/
+	cp build/SteamAuth.dll /usr/local/bin/
+	cp bash-tab-completion /etc/bash_completion.d/steamguard
+
+uninstall:
+	rm -f /usr/local/bin/steamguard
+	rm -f /usr/local/bin/Newtonsoft.Json.dll
+	rm -f /usr/local/bin/SteamAuth.dll
+	rm -f /etc/bash_completion.d/steamguard

